mosip-packet-creator/src/main/java/io/mosip/test/packetcreator/mosippacketcreator/service/PacketSyncService.java

```diff
@@ -369,8 +387,29 @@ public  Path createIDJsonFromPersona(String personaFile, String contextKey) thro
     public String syncPacketRid(String containerFile, String name,
                                 String supervisorStatus, String supervisorComment, String proc,String contextKey, String additionalInfoReqId) throws Exception {
         
-    	if(contextKey != null && !contextKey.equals("")) {
+    	RidSyncRequestData ridSyncRequestData = prepareRidSyncRequest(containerFile, name, supervisorStatus, supervisorComment, proc, contextKey, additionalInfoReqId);
+        JSONArray response = apiRequestUtil.syncRid(baseUrl,baseUrl+syncapi, ridSyncRequestData.getRequestBody(), APIRequestUtil.getUTCDateTime(ridSyncRequestData.getTimestamp()),contextKey);
+
+        return response.toString();
+    };
     
+    public RidSyncReqResponseDTO syncPacketRidRequest(String containerFile, String name, String supervisorStatus, String supervisorComment,
+			String proc, String contextKey, String additionalInfoReqId) throws Exception {
+
+    	RidSyncRequestData ridSyncRequestData = prepareRidSyncRequest(containerFile, name, supervisorStatus, supervisorComment, proc, contextKey, additionalInfoReqId);
+		loadContext(contextKey);
```

Comment:
loadContext need to be called everytime ?

Reply:
context here loaded based on context key which is a input. It is just a map with key to its objects

Reply:
Then why should we keep it as a class level variable ?